tests/loggers/test_remote_uploader_downloader.py

```diff
@@ -174,81 +182,108 @@ def test_race_with_overwrite(tmp_path: pathlib.Path, use_procs: bool, dummy_stat
         with open(tmp_path / 'samples' / f'sample_{i}', 'w+') as f:
             f.write(str(i))
 
-    # Create the object store logger
-    remote_uploader_downloader = RemoteUploaderDownloader(
-        object_store_cls=DummyObjectStore,
-        object_store_kwargs={
-            'dir': tmp_path / 'object_store_backend',
-        },
-        num_concurrent_uploads=4,
-        use_procs=use_procs,
-        upload_staging_folder=str(tmp_path / 'staging_folder'),
-        num_attempts=1,
-    )
+    # Patching does not work when using multiprocessing with spawn, so we also
+    # patch to use fork
+    fork_context = multiprocessing.get_context('fork')
```

Comment:
can I get more context into what you're doing here? we fork the process instead of spawn it to get patching to work, so in order to do that without directly changing RemoteUploaderDownloader, we patch spawn to be fork instead?

Reply:
yeah, it is basically what you said. Inside `RemoteUploaderDownloader`, multiprocessing is forced into using `spawn` instead of `fork` (https://github.com/mosaicml/composer/blob/50253c1b16646ab7c7b110237b0f851f1fe118ca/composer/loggers/remote_uploader_downloader.py#L214). `spawn` does not copy over the current state, and (as far as I could tell from googling), there is no way to get `patch` to work with multiprocessing `spawn`. So I mock the `multiprocessing.get_context` call to always use `fork` so that the patch to `S3ObjectStore` works.